forte/data/base_pack.py

```diff
@@ -455,32 +456,53 @@ def on_entry_creation(
             # Use the auto-inferred control component.
             c = self.__control_component
 
-        def entry_getter(cls: Entry, attr_name: str, field_type):
+        def entry_getter(cls: Entry, attr_name: str):
             """A getter function for dataclass fields of entry object.
             When the field contains ``tid``s, we will convert them to entry
             object on the fly.
 
             Args:
                 cls: An ``Entry`` class object.
                 attr_name: The name of the attribute.
-                field_type: The type of the attribute.
             """
+
             data_store_ref = (
                 cls.pack._data_store  # pylint: disable=protected-access
             )
             attr_val = data_store_ref.get_attribute(
                 tid=cls.tid, attr_name=attr_name
             )
-            if field_type in (FList, FDict):
+            entry_type = data_store_ref.get_entry_types(
+                cls.entry_type(), attr_name
+            )
+
+            if entry_type[0] in (FList, FDict):
```

Comment:
not sure if hardcoding `FList` and `FDict` is a good idea. 

We have some constants here. https://github.com/asyml/forte/blob/master/forte/data/ontology/ontology_code_const.py#L86

Reply:
Can I add constants here?

Reply:
I'm not sure if we can use constants from here in base_pack because it makes a partial import of `PackType` from base_pack itself.

Reply:
I have instead added a constant in `core.py` where these classes are defined.

Reply:
they should be better in the same place and nearby, probably here: https://github.com/asyml/forte/blob/master/forte/data/ontology/ontology_code_const.py#L86

Reply:
As I mentioned above, if we put it there we won't be able to use it in `base_pack`

Reply:
huh, let's see whether this workout.